libnetwork/sandbox_store.go

```diff
@@ -235,13 +229,25 @@ func (c *Controller) sandboxCleanup(activeSandboxes map[string]interface{}) {
 			var ep *Endpoint
 			if err != nil {
 				log.G(context.TODO()).Errorf("getNetworkFromStore for nid %s failed while trying to build sandbox for cleanup: %v", eps.Nid, err)
-				n = &Network{id: eps.Nid, ctrlr: c, drvOnce: &sync.Once{}, persist: true}
-				ep = &Endpoint{id: eps.Eid, network: n, sandboxID: sbs.ID}
+				ep = &Endpoint{
```

Comment:
> We should remove all sandboxes not present in activeSandboxes from the store;

So that's the part that always confuses me; what should be considered the source of truth? I recall in overlay (swarm) case, the `libkv.db` (I think that's what the file named) could be deleted if it's corrupted, and the daemon would recover from that; but maybe for other networks that's not the case.

In either case, yes, this needs cleaning up, because the current flow is confusing (and a lot of duct-tape).


w.r.t. case where we fail to cleanup something, I'm not 100% against having something to keep track of that in the "state" (e.g., we do the same for containers that failed to cleanup; we mark those as `dead` (I _think_ dead containers don't show up in `docker ps ` etc, but they are used by the daemon to cleanup on restart).